web/frontend/src/api/navigation.ts

```diff
@@ -1,14 +1,16 @@
-import type { Client } from '@viamrobotics/sdk';
-import { commonApi, navigationApi } from '@viamrobotics/sdk';
+import * as THREE from 'three';
+import { type Client, commonApi, navigationApi } from '@viamrobotics/sdk';
+import { OrientationVector } from 'three-orientation-vector';
 import { rcLogConditionally } from '@/lib/log';
+import type {
+  BoxGeometry, CapsuleGeometry, NavigationModes, Obstacle, SphereGeometry, Waypoint,
+} from './types/navigation';
+import { notify } from '@viamrobotics/prime';
 
-export type NavigationModes =
-  | typeof navigationApi.Mode.MODE_MANUAL
-  | typeof navigationApi.Mode.MODE_UNSPECIFIED
-  | typeof navigationApi.Mode.MODE_WAYPOINT
+export * from './types/navigation';
 
-export type LngLat = { lng: number, lat: number }
-export type Waypoint = LngLat & { id: string }
+const ov = new OrientationVector();
+const quat = new THREE.Quaternion();
```

Comment:
Exactly, this is a pattern that you'll see often in three.js-land (specifically declaring THREE classes at the module level whenever possible). This to prevent GC thrashing when using these classes rapidly in a temporal sense. I think I'm being too judicious here and in combo with your side effects in map comment I think I'm just going to instantiate a separate quaternion and orientation vector for each geometry. Memory-wise they're lightweight (just x, y, z, th | w props)